src/openzaak/utils/query.py

```diff
@@ -148,22 +144,22 @@ def get_filters(
                                 catalogus_authorisation.max_vertrouwelijkheidaanduiding
                             )
                         )
-                        vertrouwelijkheidaanduiding_whens.append(
-                            When(
-                                **{f"{prefix}{loose_fk_field}": instance},
-                                then=Value(choice_item_order),
-                            )
-                        )
+                        va_mapping[choice_item_order].append(instance)
+
+        # Group the When clauses by vertrouwelijkheidaanduiding, to avoid a lot of
+        # duplicate `THEN ...` statements
+        vertrouwelijkheidaanduiding_whens = [
+            When(**{f"{prefix}{loose_fk_field}__in": instances}, then=max_va)
+            for max_va, instances in va_mapping.items()
+        ]
 
         # filtering:
         # * only allow the white-listed loose-fk objects, explicitly
         # * apply the filtering to limit cases within case-types to the maximal
         #   confidentiality level
         filters = {
             f"{prefix}{loose_fk_field}__in": loose_fk_objecten,
-            "_va_order__lte": Case(
-                *vertrouwelijkheidaanduiding_whens, output_field=IntegerField()
```

Comment:
output_field not needed anymore?

Reply:
Looked over the fact that I removed this. I checked and if you don't specify the output field, django will infer the type from the when clauses, but I'll add it back to make it explicit